ci/build-mingw64.sh

```diff
@@ -16,7 +18,7 @@ export AR=$TARGET-ar
 export NM=$TARGET-nm
 export RANLIB=$TARGET-ranlib
 
-export CFLAGS="-O2 -pipe -Wall -D_FORTIFY_SOURCE=2"
+export CFLAGS="-O2 -pipe -Wall -D_FORTIFY_SOURCE=3"
```

Comment:
Why not use the one sourced from common?

Reply:
Because mingw completely breaks for inexplicable reasons if I try to use cflags from common.

Reply:
stramge

Reply:
If you're curious, I kept running into `../misc/io_utils.c:48:42: error: unknown conversion type character ‘z’ in format [-Werror=format=]` for some reason. Tried various things but same result. My only conclusion is that the flag order matters somehow but wasn't sure what exactly or why.

https://github.com/mpv-player/mpv/actions/runs/11389777584/job/31689875179

Reply:
Seems to work fine with https://github.com/kasper93/mpv/commit/1d2e1f379be495bdaf91b75e975afe5db365487a

Reply:
I tried that exact same thing earlier and it didn't work.... I'll give it a test again.

Reply:
Yup failed again. https://github.com/mpv-player/mpv/actions/runs/11393889558/job/31703114086?pr=15113

Seems like removing `-Dc_args="-Wno-error=deprecated -Wno-error=deprecated-declarations"  breaks this somehow.

Reply:
> Seems like removing `-Dc_args="-Wno-error=deprecated -Wno-error=deprecated-declarations" breaks this somehow.

meson probably ignores CFLAGS when c_args are defined. Probably we should fix that in this case.

Reply:
Yeah adding that option in manually fixes it. I don't really understand why but that's at least somewhat less duplication.

Reply:
>meson probably ignores CFLAGS when c_args are defined. Probably we should fix that in this case.

Can confirm that it does. So the CFLAGS in the job have been broken the whole time actually.

Reply:
@Dudemanguy Could we add
```
# GCC has never been updated for c99 format specifiers on mingw builds
# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=114775
export CFLAGS="$CFLAGS -Wno-error=format -Wno-error=format-extra-args"
```
in build_mingw64.sh? It is broken and I don't really care about msvcrt build. Note that it works fine on any platform we support, it just that GCC diagnostics are not updated.

Reply:
Or better, just use our wrapper to workaround this mess. We already have a "fix" for format specifiers checks in our code. Just that with fortify we get errors from broken gcc headers.

``` c
diff --git a/misc/io_utils.c b/misc/io_utils.c
index fafed772cc..777e1c780c 100644
--- a/misc/io_utils.c
+++ b/misc/io_utils.c
@@ -45,7 +45,7 @@ int mp_mkostemps(char *template, int suffixlen, int flags)
         // not truly random; just a counter would be sufficient).
         size_t fuckmess = mp_rand_next();
         char crap[7] = "";
-        snprintf(crap, sizeof(crap), "%06zx", fuckmess);
+        mp_tprintf_buf(crap, sizeof(crap), "%06zx", fuckmess);
         memcpy(t, crap, 6);
 
         int res = open(template, O_RDWR | O_CREAT | O_EXCL | flags, 0600);
diff --git a/osdep/windows_utils.c b/osdep/windows_utils.c
index 91eef62cb8..2befea2d86 100644
--- a/osdep/windows_utils.c
+++ b/osdep/windows_utils.c
@@ -143,9 +143,9 @@ static char *fmtmsg_buf(char *buf, size_t buf_size, DWORD errorID)
                              FORMAT_MESSAGE_IGNORE_INSERTS,
                              NULL, errorID, 0, buf, buf_size, NULL);
     if (!n && GetLastError() == ERROR_MORE_DATA) {
-        snprintf(buf, buf_size,
-                 "<Insufficient buffer size (%zd) for error message>",
-                 buf_size);
+        mp_tprintf_buf(buf, buf_size,
+                       "<Insufficient buffer size (%zd) for error message>",
+                       buf_size);
     } else {
         if (n > 0 && buf[n-1] == '\n')
             buf[n-1] = '\0';

```

Reply:
Finally works.